File: Tema 05 - Typescript/Frontend/src/models/Fundraiser.ts
import type { Donor } from "./Donor";

export  class Fundraiser {
  
name:	string;
dueDate?: Date; 
goalAmount?:	number;
status?:	string;
donors?:   Donor[]; 
currentAmount?:	number;  

constructor(name: string, status?: string,  dueDate?: Date, goalAmount?: number,  donors?: Donor[], currentAmount?: number ) {  
    this.name = name; 
    this.dueDate = dueDate; 
    this.goalAmount = goalAmount;
    this.donors = donors;  
    this.status = status; 
    this.currentAmount = currentAmount; 

}
}

File: Tema 07 - Clean Code/PetShelter/PetShelter.DataAccessLayer/Models/Fundraiser.cs
namespace PetShelter.DataAccessLayer.Models;

public class Fundraiser : IEntity
{
    public int Id { get; set; }
    public string Name { get; set; }


    public decimal CurrentAmount { get; set; }
    public decimal GoalAmount { get; set; }

    public string Status { get; set; }


    public DateTime CreationDate { get; set; }
    public DateTime DueDate { get; set; }

    public int  OwnerId { get; set; }   
    public Person  Owner { get; set; }

    /// <summary>
    ///     Can be used to identify the List of Donors   also besides linking donations to fundraiser. 
    /// </summary>
    public ICollection<Donation> Donations { get; set; }



}